File: angular-cooking-class/src/app/recipe.ts
export interface Recipe {
        count: number;
        results: Result[];
      }
export  interface Result {
        id?: number;
        name?: string;
        description?: string;
        instructions: Instruction[];
        nutrition?: Nutrition;
        yields: string;
        keywords: object;
        num_servings: number;
        total_time_tier: object;
        thumbnail_url: string;
        thumbnail_alt_text: string;
        total_time_minutes: object;
        original_video_url: string;
        cook_time_minutes: object;
        seo_path: object;
        seo_title: object;
        prep_time_minutes: object;
        sections: Section[];
      }
export  interface Nutrition {
        carbohydrates: number;
        fiber: number;
        updated_at: Date;
        protein: number;
        fat: number;
        calories: number;
        sugar: number;
      }
export  interface Instruction {
        id: number;
        position: number;
        display_text: string;
        start_time: number;
        appliance: object;
        end_time: number;
      }
export  interface Ingredient {
        id: number;
        name: string;
      }
export  interface Section {
        name: string;
        position: number;
        components: Component[];
      }
export  interface Component {
        id: number;
        position: number;
        raw_text: string;
        extra_comment: string;
        ingredient: Ingredient;
      }

File: CookingAppApi/CookingAppApi/Models/Recipe.cs
using System;
using System.Collections.Generic;

namespace CookingAppApi.Models;

public partial class Recipe
{
    public int RecipeId { get; set; }

    public int? Id { get; set; }

    public string? Name { get; set; }

    public string? Description { get; set; }

    public string? Yields { get; set; }

    public int? NumServings { get; set; }

    public string? ThumbnailUrl { get; set; }

    public string? ThumbnailAltText { get; set; }

    public string? OriginalVideoUrl { get; set; }

    public int? CookTimeMinutes { get; set; }

    public int? PrepTimeMinutes { get; set; }

    public int? TotalTimeMinutes { get; set; }

    public string? SeoPath { get; set; }

    public string? SeoTitle { get; set; }

    public string? TotalTimeTier { get; set; }

    public virtual ICollection<Favorite> Favorites { get; set; } = new List<Favorite>();

    public virtual ICollection<Ingredient> Ingredients { get; set; } = new List<Ingredient>();

    public virtual ICollection<Instruction> Instructions { get; set; } = new List<Instruction>();

    public virtual ICollection<Nutrition> Nutritions { get; set; } = new List<Nutrition>();

    public virtual ICollection<Section> Sections { get; set; } = new List<Section>();
}
